sale_triple_discount/models/sale_order_line.py

```diff
@@ -97,6 +97,9 @@ def triple_discount_preprocess(self):
         this method is called multiple times."""
 
         prev_values = dict()
+        digits = self._fields["discount"]._digits
+        self.env["sale.order.line"]._fields["discount"]._digits = (16, 16)
```

Comment:
I'm not fan of this. I think this could lead to unwanted side effects.

Reply:
Its not very pretty but I think it hasn't caused any issues for us so far, its mainly copying the process in the account equivalent module https://github.com/OCA/account-invoicing/blob/d2e12c03b50011bb788cf482346710f0fefe0de3/account_invoice_triple_discount/models/account_move_line.py#L73-L74 so it calculates the same discount, otherwise it can lead to different rounding